src/pcm.c

```diff
@@ -1806,6 +1806,9 @@ int pcm_ioctl(struct pcm *pcm, int request, ...)
     arg = va_arg(ap, void *);
     va_end(ap);
 
-    // FIXME Does not handle plugins
+#ifdef TINYALSA_USES_PLUGINS
+    return pcm->ops->ioctl(pcm->data, request, arg);
```

Comment:
pcm->ops->ioctl() is valid for pcm_hw.c as well. There is no need to have #ifdef check.

Reply:
Can we also add support of pcm->ops->ioctl() at other places in the file.
e.g.
    if (ioctl(pcm->fd, SNDRV_PCM_IOCTL_DELAY, &pcm->pcm_delay) < 0)
        return -1;